runtime/compiler/x/env/J9CPU.hpp

```diff
@@ -66,12 +66,14 @@ class OMR_EXTENSIBLE CPU : public J9::CPU
    CPU(const OMRProcessorDesc& processorDescription) : J9::CPU(processorDescription) {}
 
 public:
+   
+   static TR::CPU detectRelocatable(OMRPortLibrary * const omrPortLib);
```

Comment:
This is a static. Does this work with extensible classes?
I don't see `detectRelocatable()` being called anywhere in this changeset. It is called in `OMR::CompilerEnv::initializeRelocatableTargetEnvironment()` but will use its own definition of `detectRelocatable` I think.

Reply:
I think `static` works with extensible classes. It's proven to work with the `detect()` set of changes. ` TR::CPU::detectRelocatable(TR::Compiler->omrPortLib);` will use the most derived definition (Notice the TR::CPU). So once `J9::CPU::detectRelocatable()` is defined, it will start using `J9::CPU::detectRelocatable()` instead of `OMR::CPU::detectRelocatable`.

Reply:
I verified that `detectRelocatable()` does not seem to be the cause of the issue i mentioned. It's 
```cpp
   if (((TR_J9VMBase *)fe)->isAOT_DEPRECATED_DO_NOT_USE())
      _target = TR::Compiler->relocatableTarget;
```

I failed to see why this would create the javac being jammed problem though.

Reply:
```cpp
-      _target = TR::Compiler->relocatableTarget;
+      _target = TR::Compiler->target;
```

If i take out relocatableTarget then it works. Something is wrong with relocatableTarget..

Reply:
If you replace the body of `J9::X86::CPU::detectRelocatable` with the body of `OMR::CPU::detectRelocatable` and leave `_target = TR::Compiler->relocatableTarget;`, do you still see the problem?

Reply:
Yes I tried it. I also tried removing `J9::X86::CPU::detectRelocatable` entirely but leaving `_target = TR::Compiler->relocatableTarget`. still see the problem.

Reply:
Trying my last attempt to make sure `J9::X86::CPU::detectRelocatable` works: instead of doing `cpu.target = TR::CPU::detect()`, i'm trying `cpu.target = TR::CPU::detectRelocatable()` and avoid using relocatableTarget at all. If this works we can conclude that `J9::X86::CPU::detectRelocatable` is working properly.

Reply:
And it worked. So it's definitely `TR::Compiler->relocatableTarget`..
I can try the other combo `cpu.relocatableTarget = TR::CPU::detect()` and use relocatableTarget instead of target everywhere. I'm guessing that will fail.

Reply:
To my surprise the above configuration worked.

Reply:
as soon as we do this:
```cpp
-   _target(TR::Compiler->target),
+   _target(TR::Compiler->relocatableTarget),
```
(In OMRCompilation constructor)
It stops working. There is some thing different between `TR::Compiler->target` and `TR::Compiler->relocatableTarget` that was not intended.

Reply:
I think I'm starting to understand why
After the initialization, we continue to set things for the target:
```cpp
 5180  #if !defined(J9_PROJECT_SPECIFIC)
 5181     TR::Compiler->host.setNumberOfProcessors(2);
 5182:    TR::Compiler->target.setNumberOfProcessors(2);
 5183     TR::Compiler->host.setSMP(true);
 5184:    TR::Compiler->target.setSMP(true);
 5185  #endif
```
We need to do the same for relocatableTarget as well if we want the same behavior. Or I should look into sinking them down to the constructor (which most likely not possible).